Бугров Никита

Rinsz opened the pull request:
@PhonkX

Rinsz committed:
TagsCloudVisualization first task CircularCloudLayouter implemented

Rinsz committed:
TagsCloudVisualization first task tests created

Rinsz committed:
Task 2. CircularCloudLayouter.cs Image Creator added

Rinsz committed:
Task 2. Program.cs random generator added

Rinsz committed:
Task 2. README.md added

Rinsz committed:
Task 3. Image providing TearDown added

Rinsz committed:
CircularCloudLayouter_Should uncompleted test removed

PhonkX commented on cs/Samples/Samples.csproj:
```diff
@@ -10,8 +10,9 @@
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.7.1" />
-    <PackageReference Include="NUnit" Version="3.12.0" />
-    <PackageReference Include="NUnit3TestAdapter" Version="3.17.0" />
+    <PackageReference Include="NUnit" Version="3.10.0" />
```
А зачем версии пакетов откатил?

Rinsz replied:
Тесты приняли ислам)
Понятия не имею почему у меня студия не хочет их заводить на последних версиях.

PhonkX replied:
С решарпером запускаешь? Нужно его обновить, у меня коллеги сталкивались с таким поведением

Rinsz replied:
Ага. Хорошо, попробую на обновления проверить

PhonkX commented on cs/TagsCloudVisualization/CircularCloudLayouter.cs:
```diff
@@ -0,0 +1,123 @@
+            potentialPosingPoints.AddRange(GetPotentialPosingPointsFromRectangle(rect));
+            isFirst = false;
+            return rect;
+
```
Лишняя пустая строчка

PhonkX commented on cs/TagsCloudVisualization/CircularCloudLayouter.cs:
```diff
@@ -0,0 +1,123 @@
+                }
+            }
+            
+            throw new ArithmeticException("Unknown error");
```
Кажется, тут очень даже понятная ошибка - не хватило места

Rinsz replied:
Хм, ну, я в принципе раздумывал такую идею, что юзер могу бы указать размер области, для начала. Да и обычно такие штуки выводятся в векторной графике. Просто, расположить то можно, но увидеть это можно будет только в случае, если расширить размер битмапки.

Или речь о памяти?)

PhonkX commented on cs/TagsCloudVisualization/CircularCloudLayouter.cs:
```diff
@@ -0,0 +1,123 @@
+            if (isFirst)
+                return PutFirstRectangle(rectangleSize);
+
+            var rect = ChangeRectanglePosition(new Rectangle(Center, rectangleSize));
```
Скорее, PlaceRectangle

Rinsz replied:
Действительно)
Забыл сменить название, концепция менялась. Поправлюю

PhonkX commented on cs/TagsCloudVisualization/CircularCloudLayouter.cs:
```diff
@@ -0,0 +1,123 @@
+            return rect;
+        }
+
+        public void CreateImage(string path, string fileName)
```
Попробуй добиться того, чтобы этот метод оказался в специальной сохранялке

Rinsz replied:
Имеется ввиду вынести его в отдельный класс и обработать соответствующе?

PhonkX replied:
Да

PhonkX commented on cs/TagsCloudVisualization/TagsCloudVisualization.csproj:
```diff
@@ -0,0 +1,91 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Для интереса попробуй создать проект с target framework netcoreapp3.1, посмотри на отличия csproj'ей.

Rinsz committed:
CircularCloudLayouter optimization

-Posing points which impossible to use are deleting now

-ChangeRectanglePosition renamed to PlaceRectangle

Rinsz committed:
CircularCloudLayouter_Should add performance test

Rinsz committed:
ImageCreator allocated in a separate class

Rinsz committed:
CircularCloudLayouter removed unnecessary Exception

Rinsz committed:
CircularCloudLayouter fix
